src/legacy/ui/public/saved_objects/components/saved_object_finder.tsx

```diff
@@ -248,73 +328,169 @@ class SavedObjectFinder extends React.Component<SavedObjectFinderProps, SavedObj
+      </EuiFlexGroup>
+    );
+  }
 
-        {this.props.callToActionButton && (
-          <EuiFlexItem grow={false}>{this.props.callToActionButton}</EuiFlexItem>
-        )}
+  private renderLoadingIndicator() {
+    return (
+      <EuiFlexGroup justifyContent="center">
+        <EuiFlexItem grow={false}>
+          <EuiSpacer />
+          <EuiLoadingSpinner data-test-subj="savedObjectFinderLoadingIndicator" />
+        </EuiFlexItem>
       </EuiFlexGroup>
     );
   }
 
-  private renderTable() {
-    const pagination = {
-      pageIndex: this.state.page,
-      pageSize: this.state.perPage,
-      totalItemCount: this.state.items.length,
-      hidePerPageOptions: Boolean(this.props.fixedPageSize),
-      pageSizeOptions: [5, 10, 15],
-    };
-    // TODO there should be a Type in EUI for that, replace if it exists
-    const sorting: { sort?: EuiTableCriteria['sort'] } = {};
-    if (this.state.sortField) {
-      sorting.sort = {
-        field: this.state.sortField,
-        direction: this.state.sortDirection,
-      };
-    }
-    const tableColumns = [
-      {
-        field: 'title',
-        name: i18n.translate('common.ui.savedObjects.finder.titleLabel', {
-          defaultMessage: 'Title',
-        }),
-        sortable: true,
-        render: (title: string, record: SimpleSavedObject<SavedObjectAttributes>) => {
-          const { onChoose, makeUrl } = this.props;
-
-          if (!onChoose && !makeUrl) {
-            return <span>{title}</span>;
-          }
-
-          return (
-            <EuiLink
-              onClick={
-                onChoose
-                  ? () => {
-                      onChoose(record.id, record.type);
-                    }
-                  : undefined
-              }
-              href={makeUrl ? makeUrl(record.id) : undefined}
-              data-test-subj={`savedObjectTitle${title.split(' ').join('-')}`}
-            >
-              {title}
-            </EuiLink>
-          );
-        },
-      },
-    ];
+  private renderListing() {
     const items = this.state.items.length === 0 ? [] : this.getPageOfItems();
+    const { onChoose, savedObjectMetaData } = this.props;
+
     return (
-      <EuiBasicTable
-        items={items}
-        loading={this.state.isFetchingItems}
-        columns={tableColumns}
-        pagination={pagination}
-        sorting={sorting}
-        onChange={this.onTableChange}
-        noItemsMessage={this.props.noItemsMessage}
-      />
+      <>
+        {this.state.isFetchingItems ? (
+          this.renderLoadingIndicator()
+        ) : items.length > 0 ? (
+          <FixedEuiListGroup maxWidth={false}>
+            {items.map(item => {
+              const currentSavedObjectMetaData = savedObjectMetaData.find(
+                metaData => metaData.type === item.type
+              );
+              const fullName = `${item.title} (${currentSavedObjectMetaData!.name})`;
```

Comment:
Since visualization objects icons are actually this vis type icon and not just the visualization app icon, can we append the vis type to the title?

<img width="375" alt="screen shot 2019-03-07 at 14 54 05 pm" src="https://user-images.githubusercontent.com/549577/53985212-12e3fe00-40e9-11e9-96c4-dced1cbc0149.png">

In that example, it would read something like

> [eCommerce] Average Sold Quantity (Gauge visualization)